File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    map<int, int> count;
    for (int value : tangerine) {
        count[value]++;
    }
    
    vector<int> counts;
    
    for (auto& [value, cnt] : count) {
        counts.push_back(cnt);
    }
    
    sort(counts.rbegin(), counts.rend());
    
    for (int c : counts) {
        k -= c;
        answer++;
        
        if (k <= 0) {
            break;
        }
    }
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    
    count = {}

    for value in tangerine:
        if value in count:
            count[value] += 1
        else:
            count[value] = 1
    
    counts = sorted(count.values(),reverse=True)
    
    for c in counts:
        k -= c
        answer += 1
    
        if k <= 0:
            break
    
    return answer